kong/plugins/ip-restriction/handler.lua

```diff
@@ -52,31 +53,64 @@ local function match_bin(list, binary_remote_addr)
 end
 
 
-function IpRestrictionHandler:access(conf)
+local function do_exit(status, message, is_http)
+  if is_http then
+    return kong.response.error(status, message)
+
+  else
+    local tcpsock, err = ngx.req.socket(true)
+    if err then
+      error(err)
+    end
+
+    tcpsock:send(cjson.encode({
```

Comment:
We should localize `cjson.encode` at the module level.

Reply:
Done

Reply:
I think you misunderstand my meaning. The code should be:
```
local cjson_encode = cjson.encode
```